lib/galaxy/security/validate_user_input.py

```diff
@@ -46,6 +46,13 @@ def validate_email(trans, email, user=None, check_dup=True, allow_empty=False):
             domain = ('.').join(domain.split('.')[-2:])
         if domain in trans.app.config.blacklist_content:
             message = "Please enter your permanent email address."
+    #  If the whitelist is not empty filter out any domain not in the list.
+    elif trans.app.config.whitelist_content is not None:
```

Comment:
So with `elif` here, these will be mutually exclusive options, if blacklist file is set, whitelist will be ignored. This sounds reasonable but should be documented.

Reply:
Thanks for the reminder! I am going to add to the documentation thatif blacklist is defined then whitelist is ignored.

Reply:
@hexylena they *are* mutually exclusive, but shouldn't it be the other way round - if whitelist is set, ignore the blacklist? (see discussion further down the page)